projects/solar/test/test_sense_temperature.c

```diff
@@ -72,21 +77,23 @@ void test_single_thermistor_cycle(void) {
   TEST_ASSERT_EQUAL(adc_channel, s_adc_channel_passed);
   data_store_get_is_set(DATA_POINT_TEMPERATURE(0), &is_set);
   TEST_ASSERT_TRUE(is_set);
-  data_store_get(DATA_POINT_TEMPERATURE(0), &set_value);
-  TEST_ASSERT_EQUAL(TEST_ADC_RAW_READING, set_value);
+  data_store_get_signed(DATA_POINT_TEMPERATURE(0), &set_value);
+  TEST_ASSERT_EQUAL(TEST_RTD_CONVERTED_READING, set_value);
 }
 
 // Test that everything works correctly for a single cycle with the max number of thermistors.
 void test_max_thermistors_cycle(void) {
   bool is_set;
-  uint32_t set_value;
+  int32_t set_value;
   SenseTemperatureSettings settings = {
     .num_thermistors = MAX_THERMISTORS,
   };
   // take the pins sequentially from PA0 on
   for (uint8_t thermistor = 0; thermistor < MAX_THERMISTORS; thermistor++) {
-    settings.thermistor_pins[thermistor].port = thermistor / GPIO_PINS_PER_PORT;
-    settings.thermistor_pins[thermistor].pin = thermistor % GPIO_PINS_PER_PORT;
+    settings.thermistor_settings[thermistor].thermistor_address.port =
+        thermistor / GPIO_PINS_PER_PORT;
+    settings.thermistor_settings[thermistor].thermistor_address.pin =
+        thermistor % GPIO_PINS_PER_PORT;
```

Comment:
You should also set the `thermistor_type` to various values, maybe with a `thermistor % NUM_THERMISTOR_TYPES` expression.